### PR TITLE
For #17917: Use View binding in tabs tray

### PR DESCRIPTION
**Pull request by codrut-topliceanu**
For https://github.com/mozilla-mobile/fenix/issues/17917

### Pull Request checklist

- [x] **Tests**: This PR includes no tests as it is only a minor change
- [x] **Screenshots**: This PR does not include screenshots as no user visible changes have been made.
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md).

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFragment.kt [hunk 1]
```diff
@@ -88,8 +93,21 @@ class TabsTrayFragment : AppCompatDialogFragment() {
...
         savedInstanceState: Bundle?
     ): View {
-        val containerView = inflater.inflate(R.layout.fragment_tab_tray_dialog, container, false)
-        inflater.inflate(R.layout.component_tabstray2, containerView as ViewGroup, true)
+        _tabsTrayDialogBinding = FragmentTabTrayDialogBinding.inflate(
+            inflater,
+            container,
+            false
+        )
+        _tabsTrayBinding = ComponentTabstray2Binding.inflate(
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFragment.kt:101]: I wouldn't be opposed to removing the '2' in the xml class names if you're touching this code. 👀

Comment (codrut-topliceanu) [reply app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFragment.kt:101]: But then how will we know of our history when we reach component_tabstray3.xml ? 😆

**Review by Mugurell — DISMISSED** [review]
LGTM.

**Review by Mugurell — APPROVED** [review]
CI is green after a restart with no other changes so refreshing my r+.

**Comment by Mugurell** [comment]
```
androidx.test.espresso.PerformException: Error performing 'single click' on view 'with id: org.mozilla.fenix.debug:id/trackingProtectionDetails'.
at androidx.test.espresso.PerformException$Builder.build(PerformException.java:5)
at androidx.test.espresso.base.DefaultFailureHandler.getUserFriendlyError(DefaultFailureHandler.java:25)
at androidx.test.espresso.base.DefaultFailureHandler.handle(DefaultFailureHandler.java:36)
at androidx.test.espresso.ViewInteraction.waitForAndHandleInteractionResults(ViewInteraction.java:106)
at androidx.test.espresso.ViewInteraction.desugaredPerform(ViewInteraction.java:43)
at androidx.test.espresso.ViewInteraction.perform(ViewInteraction.java:94)
at org.mozilla.fenix.helpers.ViewInteractionKt.click(ViewInteraction.kt:18)
at org.mozilla.fenix.ui.robots.EnhancedTrackingProtectionRobot$Transition.openDetails(EnhancedTrackingProtectionRobot.kt:81)
at org.mozilla.fenix.ui.StrictEnhancedTrackingProtectionTest.testStrictVisitSheetDetails(StrictEnhancedTrackingProtectionTest.kt:155) 
```

**Pull request merged**